Sources/Common/Core/Math/index.js

```diff
@@ -383,7 +375,7 @@ export function gaussianWeight(mean, variance, position) {
 export function outer2D(x, y, out_2x2) {
   for (let i = 0; i < 2; i++) {
     for (let j = 0; j < 2; j++) {
-      out_2x2[i][j] = x[i] * y[j];
+      out_2x2[i * 2 + j] = x[i] * y[j];
```

Comment:
replace nested for loops with hardcoded exhaustive assignment (can be done in a separate PR)

Reply:
Change was made at that place.